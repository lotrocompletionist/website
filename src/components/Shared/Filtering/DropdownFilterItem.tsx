import * as React from "react";
import { DropdownItem } from "bloomer";

export interface DropdownFilterItemProps {
  id: string;
  value: string;
  onChange: React.ChangeEventHandler<HTMLInputElement>;
}

export function DropdownCheckboxFilterItem(props: DropdownFilterItemProps) {
  return (
    <DropdownItem>
      <input
        id={props.id}
        className="is-checkradio"
        type="checkbox"
        value={props.value}
        onChange={props.onChange}
        name={props.value}
      />
      <label htmlFor={props.id}>{props.value}</label>
    </DropdownItem>
  );
}
