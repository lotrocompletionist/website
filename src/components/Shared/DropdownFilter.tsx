import * as React from "react";
import {
  Dropdown,
  DropdownTrigger,
  Button,
  DropdownMenu,
  Icon,
  DropdownContent,
  DropdownDivider
} from "bloomer";
import { DropdownFilterItemProps } from "./DropdownFilterItem";

interface DropdownFilterProps {
  header: string;
  values: string[];
  onChange: React.ChangeEventHandler<HTMLInputElement>;
  filterItem: React.StatelessComponent<DropdownFilterItemProps>;
}

interface DropdownFilterState {
  isActive: boolean;
}

export class DropdownFilter extends React.Component<
  DropdownFilterProps,
  DropdownFilterState
> {
  constructor(props: DropdownFilterProps) {
    super(props);
    this.state = { isActive: false };
  }

  onClickNav = () => {
    this.setState({ isActive: !this.state.isActive });
  };
  render() {
    return (
      <Dropdown isActive={this.state.isActive}>
        <DropdownTrigger onClick={this.onClickNav}>
          <Button isOutlined aria-haspopup="true" aria-controls="dropdown-menu">
            <span>{this.props.header}</span>
            <Icon icon="angle-down" isSize="small" />
          </Button>
        </DropdownTrigger>
        <DropdownMenu>
          <DropdownContent>
            {this.props.values.map(value => (
              <this.props.filterItem
                key={value}
                value={value}
                onChange={this.props.onChange}
              />
            ))}
          </DropdownContent>
        </DropdownMenu>
      </Dropdown>
    );
  }
}
