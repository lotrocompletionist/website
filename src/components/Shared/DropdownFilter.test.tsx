import * as React from "react";
import { create } from "react-test-renderer";
import { DropdownFilter } from "./DropdownFilter";
import { DropdownFilterItemProps } from "./DropdownFilterItem";

test("<DropdownFilter> renders correctly", () => {
  const header = "Regions";
  const values = ["Ered-Luin", "The Shire", "The North Downs"];
  const onChange = (event: React.ChangeEvent<HTMLInputElement>) => {};
  const filterItem = (props: DropdownFilterItemProps) => <div />;

  const tree = create(
    <DropdownFilter
      header={header}
      values={values}
      onChange={onchange}
      filterItem={filterItem}
    />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
