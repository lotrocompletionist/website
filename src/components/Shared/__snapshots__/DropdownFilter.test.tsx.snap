// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DropdownFilter> renders correctly 1`] = `
<div
  className="dropdown"
>
  <div
    className="dropdown-trigger"
    onClick={[Function]}
  >
    <button
      aria-controls="dropdown-menu"
      aria-haspopup="true"
      className="button is-outlined"
      type="button"
    >
      <span>
        Regions
      </span>
      <span
        className="icon is-small"
      >
        <span
          aria-hidden="true"
          className="fa fa-angle-down"
        />
      </span>
    </button>
  </div>
  <div
    className="dropdown-menu"
  >
    <div
      className="dropdown-content"
    >
      <div />
      <div />
      <div />
    </div>
  </div>
</div>
`;
